pymc/distributions/distribution.py

```diff
@@ -623,12 +697,7 @@ def dist(
             logcdf = default_not_implemented(class_name, "logcdf")
 
         if moment is None:
-            moment = functools.partial(
-                default_moment,
-                rv_name=class_name,
-                has_fallback=True,
-                ndim_supp=ndim_supp,
-            )
+            moment = dist_moment
```

Comment:
Remove this if statement (see other comment about default)

Reply:
@ricardoV94  Hm, I think need this condition to avoid overriding moment provided by user, how can we avoid it without condition?

Reply:
More specificaly, If I try to remove  if statement `TestCustomSymbolicDist::test_custom_methods` fails

Reply:
There are two levels. We dispatch the general moment to the parent base class. Whenever a new subclass is created here and the user provided a moment we register it on the subclass (so it has preference). If the user didn't provide anything, we don't register and the parent class one will be used.